apis/projectcontour/v1/httpproxy.go

```diff
@@ -379,6 +385,15 @@ type UpstreamValidation struct {
 	SubjectName string `json:"subjectName"`
 }
 
+// DownstreamValidation defines how to verify the client certificate.
+type DownstreamValidation struct {
```

Comment:
This is just a struct name so no big deal, but do we need to give it an identity? What if we used this to validate connections elsewhere? Maybe a more generic name that could allow for re-use later?

Reply:
Could you explain bit more what you mean by giving an identity?  I think I didn't understand it :)

There was `UpstreamValidation` before and `DownstreamValidation` mirrors that with the exception that `SubjectName` attibute is not present. Otherwise single generic structure would have worked.

Reply:
The way I was thinking about this was that as we add more API surface, we would eventually converge the two struct names into a common `PeerValidation` like the one we use internally.

Reply:
Yeah, what I meant was @jpeach mentioned. Calling the struct something more generic so that it could be reused for other similar items later.

Reply:
@stevesloka Were you thinking that we should make that change now? If we do, that would imply to API users that all the fields currently in `UpstreamValidation` would work, or we would have to add extra validation code to display errors if those fields were used in a downstream context.

Reply:
I never said merge it with UpstreamValidation. Just ship as is. LGTM

Reply:
@jpeach  ^^

Reply:
'kk thanks @stevesloka . I'll resolve this conversation and we can revisit. These names don't affect API compatibility, so we can change them later if we like.